### PR TITLE
Use base dev images

### PR DESCRIPTION
**Pull request by quartje**
This will use images from the OpenConext BaseContainer project. The :dev images in the project repositories are redundant.

I've also included a way to be able to override a dev image with another image by overriding it in the file ".env". This allows a developer to choose which container is used for developing, useful when migration to another development container (eg in the case of upgrading to PHP 8.2)

### CODE REVIEW

File: stepup/README.md [hunk 1]
```diff
@@ -73,3 +73,7 @@ To mount the code in multiple containers:
 `start-dev-env.sh webauthn:/home/dan/Stepup-webauthn gateway:/home/dan/Stepup-gateway`
 You can add as many services+local code paths that you need.
 The recommended way is to use absolute paths and the script requires the name of the service and local code path to be separated by a `:`, for each service.
+
+# PHP 8.2 for development
+The default development container is based on the base image with PHP7.2. You can override this on a per service basis. Uncomment the appropiate line for this in the file ".env" so it uses the PHP8.2 container
```
Comment (MKodde) [inline stepup/README.md:78]: :heart:

File: stepup/demogssp/docker-compose.override.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+services:
+  demogssp:
+    image: ghcr.io/openconext/openconext-basecontainers/${DEMOGSSP_PHP_IMAGE:-php72-apache2-node14-composer2:latest}
+    volumes:
+      - ${DEMOGSSP_CODE_PATH}:/var/www/html
```
Comment (MKodde) [inline stepup/demogssp/docker-compose.override.yml:5]: Thanks for adding this override. Had that one as an uncommited change on my local machine.

One consideration. The mother docker-compose still sets the APP_ENV=prod for this service. When you start the dev override. The env var is not overridden.

Please remove the APP_ENV reference(s) from the base `/stepup/docker-compose.yml`. I see that was also the case for other apps.

**Review by MKodde — CHANGES_REQUESTED** [review]
Super nice work @quartje 

See my one suggestion

File: stepup/.env [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#AZUREMFA_PHP_IMAGE=php82-apache2-node20-composer2:latest
+#DEMOGSSP_PHP_IMAGE=php82-apache2-node20-composer2:latest
```
Comment (parijke) [inline stepup/.env:2]: What is the difference between these images? But if it is meant that we CAN use different images for different projects, I understand. For now they seem the same

**Review by parijke — APPROVED** [review]

File: stepup/.env (hunk 3 above)
Comment (quartje) [reply stepup/.env:2]: That's correct, you can override it on a per project basis.

File: stepup/gateway/docker-compose.override.yml [hunk 4]
```diff
@@ -1,5 +1,7 @@
 services:
   gateway:
-    image: ghcr.io/openconext/stepup-gateway/stepup-gateway:dev
+    image: ghcr.io/openconext/openconext-basecontainers/${GATEWAY_PHP_IMAGE:-php72-apache2-node14-composer2:latest}
     volumes:
       - ${GATEWAY_CODE_PATH}:/var/www/html
+    environment:
+      - APP_ENV=dev
```
Comment (parijke) [inline stepup/gateway/docker-compose.override.yml:7]: @MKodde does we have to add ` APP_DEBUG: true` to that?

File: stepup/ra/docker-compose.override.yml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 services:
   ra:
-    image: ghcr.io/openconext/stepup-ra/stepup-ra:dev
+    image: ghcr.io/openconext/openconext-basecontainers/${RA_PHP_IMAGE:-php72-apache2-node14-composer2:latest}
     volumes:
       - ${RA_CODE_PATH}:/var/www/html
```
Comment (MKodde) [inline stepup/ra/docker-compose.override.yml:5]: No `APP_ENV=dev` addition here @quartje ?

File: stepup/gateway/docker-compose.override.yml (hunk 4 above)
Comment (MKodde) [reply stepup/gateway/docker-compose.override.yml:7]: @quartje Pauls suggestion seems to make sense to me. Why not enable the debugger while we're at it?

File: stepup/ra/docker-compose.override.yml (hunk 5 above)
Comment (quartje) [reply stepup/ra/docker-compose.override.yml:5]: Ah I forgot this one. FTFY.

File: stepup/gateway/docker-compose.override.yml (hunk 4 above)
Comment (quartje) [reply stepup/gateway/docker-compose.override.yml:7]: sure. I've added it!

**Pull request merged**